Automattic/wp-calypso
# Jetpack Connect: Co-brand authorization form header for partners

With Jetpack Start (JPS) v2, we will be sending users through the standard Jetpack Connect (JPC) flow, as opposed to sending users through oAuth. Because of this, we need to add the ability to co-brand the JPC flow when the user is connecting from a hosting partner.

Working with @jeffgolenski, this is what we've come up with:

<img width="481" alt="screen shot 2017-10-11 at 3 53 49 pm" src="https://user-images.githubusercontent.com/1126811/31466348-7ab399ae-ae9c-11e7-9db4-d203c789331b.png">
<img width="480" alt="screen shot 2017-10-11 at 3 34 21 pm" src="https://user-images.githubusercontent.com/1126811/31466349-7acdebe2-ae9c-11e7-87dc-1a80f68700c4.png">
<img width="466" alt="screen shot 2017-10-11 at 3 35 17 pm" src="https://user-images.githubusercontent.com/1126811/31466350-7aebe070-ae9c-11e7-81d2-b405c0742545.png">

To test:

- Checkout PR
- Go to Jetpack site
- Disconnect site
- Click "Connect Jetpack" button
- Before the URL rewrites, copy it
- Replace `wordpress.com` with `calypso.localhost:3000`
- Add `&partner_id=51946` or `&partner_id=49640`
- Verify that you see co-branded form header
- Remove `&partner_id={...}`
- Verify that co-branding is removed
